quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManager.java

```diff
@@ -363,76 +363,76 @@ private static void processConfigGroup(IJavaElement javaElement, String baseKey,
-									// The key Map must be a String
-									processMap(field, subKey, rawTypeParameters[1], docs, location, source, configPhase,
-											converter, javadocCache, quarkusProperties, monitor);
-								}
-							} else if (isList(fieldTypeName)) {
-								addField(propertyName, type, defaultValue, docs, location, source, enumerations,
-										configPhase, quarkusProperties);
-							} else if (isOptionnal(fieldTypeName)) {
-								ExtendedConfigDescriptionBuildItem item = addField(propertyName, type, defaultValue,
-										docs, location, source, enumerations, configPhase, quarkusProperties);
-								item.setRequired(false);
-							} else {
-								addField(propertyName, type, defaultValue, docs, location, source, enumerations,
-										configPhase, quarkusProperties);
-							}
+							addField(field, fieldTypeName, fieldClass, subKey, defaultValue, converter, javadocCache,
+									configPhase, quarkusProperties, monitor);
 						}
 					}
 				}
 			}
 		}
 	}
 
+	private static void addField(IField field, String fieldTypeName, IType fieldClass, String propertyName,
+			String defaultValue, DocumentationConverter converter, Map<IPackageFragmentRoot, Properties> javadocCache,
+			ConfigPhase configPhase, List<ExtendedConfigDescriptionBuildItem> quarkusProperties,
+			IProgressMonitor monitor) throws JavaModelException {
+
+		// Class type
+		String type = fieldClass != null ? fieldClass.getFullyQualifiedName() : fieldTypeName;
+
+		// Javadoc
+		String docs = getJavadoc(field, javadocCache, monitor);
+		docs = converter.convert(docs);
+
+		// Location (JAR, src)
+		IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) field.getAncestor(IJavaElement.PACKAGE_FRAGMENT_ROOT);
+		String location = packageRoot.getPath().toString();
+
+		// field and class source
+		String source = field.getDeclaringType().getFullyQualifiedName() + "#" + field.getElementName();
+
+		// Enumerations
+		List<String> enumerations = getEnumerations(fieldClass);
+
+		// Default value for primitive type
+		if ("boolean".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "false" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("int".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("long".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("double".equals(fieldTypeName)) {
+			addField(propertyName, type, ConfigItem.NO_DEFAULT.equals(defaultValue) ? "0" : defaultValue, docs,
+					location, source, enumerations, configPhase, quarkusProperties);
+		} else if ("float".equals(fieldTypeName)) {

```

Comment:
can be 1 if condition grouping the 4 cases

Reply:
```java
if (isNumber(fieldTypeName)) {...}`
```

isNumber can be basically implemented as Array.asList("short","int","long","double","float").contains(fieldTypeName)

Reply:
@fbricon I agree with you, but I have done this choice (not group the 4 cases) to have similar code of the Quarkus ConfigDefinition https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java#L257

If you think we should have even 1 condition, I will update it.

Reply:
"short" type is missing still

Reply:
That's strange teh Quarkus ConfigDefinition doesn't manage this case?

Reply:
you're right, I didn't check

Reply:
@fbricon do I must change something?

Reply:
I'd still prefer 1 condition as it's less prone to errors when maintaining the code

Reply:
@fbricon fixed